src/overlay/FlowControlCapacity.cpp

```diff
@@ -97,6 +100,15 @@ FlowControlByteCapacity::canRead() const
     return true;
 }
 
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+void
+FlowControlByteCapacity::handleTxSizeIncrease(uint64_t increase)
+{
+    mCapacity.mFloodCapacity += increase;
```

Comment:
Do we need to check for overflow here since this increase value is coming from another node and could potentially be malicious?

Reply:
so here the increase is coming from the local node's upgrade logic, which I assume won't overflow (if the network upgrades to those types of values, then probably something is really wrong anyway). We do send another SEND_MORE message with `increase` in it, but we already handle potential overflow here (the mitigation is that the problematic peer is dropped) https://github.com/stellar/stellar-core/blob/0eab126b4343cf7d93af2cc9ca97c5bf7728ee4a/src/overlay/FlowControl.cpp#L340

Reply:
Looks good. Sorry, didn't realize this overflow check was already there!